packages/store-ui/src/organisms/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,92 @@
+import type { PropsWithChildren } from 'react'
+import type { SwipeableProps } from 'react-swipeable'
+import React from 'react'
+
+import Button from '../../atoms/Button'
+import Icon from '../../atoms/Icon'
+import { RightArrowIcon, LeftArrowIcon } from './Arrows'
+import { useCarousel } from './hooks/useCarousel'
+import { useSlideVisibility } from './hooks/useSlideVisibility'
+import Bullets from '../../molecules/Bullets'
+
+export interface CarouselProps {
+  testId?: string
+  itemsPerPage?: number
+  swipeableConfigOverrides?: SwipeableProps
+}
+
+function Carousel({
+  testId = 'store-carousel',
+  itemsPerPage = 1,
+  swipeableConfigOverrides,
+  children,
+}: PropsWithChildren<CarouselProps>) {
+  const numberOfSlides = React.Children.count(children)
+  const numberOfPages = Math.ceil(numberOfSlides / itemsPerPage)
+
+  const { handlers, slide, carouselState, carouselDispatch } = useCarousel({
+    totalItems: numberOfSlides,
+    itemsPerPage,
+    swipeableConfigOverrides,
+  })
+
+  const { shouldRenderItem, isItemVisible } = useSlideVisibility({
+    itemsPerPage: carouselState.itemsPerPage,
+    currentSlide: carouselState.currentSlide,
+  })
+
+  return (
+    <section
+      data-store-carousel
+      data-testid={testId}
+      aria-label="carousel"
+      {...handlers}
+    >
+      <div data-carousel-track-container>
+        <div data-carousel-track>
+          {React.Children.map(children, (child, idx) => (
+            <div
+              key={idx}
+              data-carousel-item
+              data-visible={isItemVisible(idx) || undefined}
+            >
+              {shouldRenderItem(idx) ? child : <div />}
```

Comment:
Why need this empty `div`?  Can you explain why?

Reply:
For no particular reason, rendering `null` shouldn't be a problem here.

Reply:
I think this is because of not rendering the whole children array to the html, however, still rendering a div defeats the purpose, because it still increases the final html in SSR